src/mem/globalalloc.h

```diff
@@ -6,55 +6,56 @@
 namespace snmalloc
 {
   template<class SharedStateHandle>
-  inline static void aggregate_stats(SharedStateHandle handle, Stats& stats)
+  inline static void aggregate_stats(Stats& stats)
   {
-    auto* alloc = Pool<CoreAllocator<SharedStateHandle>>::iterate(handle);
+    auto* alloc = Pool<CoreAllocator<SharedStateHandle>>::template iterate<
```

Comment:
Should we have some `static_assert`s about `SharedStateHandle::Options.CoreAllocIsPoolAllocated` in this file, so that incorrect usage has a better error message?